Description: Use defaults file from /etc/default/ejabberd
Bug-Github: https://github.com/processone/ejabberd/pull/248
Applied-Upstream: ?
Last-Update: 2014-07-11
--- a/ejabberdctl.template
+++ b/ejabberdctl.template
@@ -46,7 +46,8 @@
         --node) ERLANG_NODE_ARG=$1 ; shift ;;
         --config-dir) ETC_DIR="$1" ; shift ;;
         --config) EJABBERD_CONFIG_PATH="$1" ; shift ;;
-        --ctl-config) EJABBERDCTL_CONFIG_PATH="$1" ; shift ;;
+        --default) EJABBERDCTL_CONFIG_PATH="$1" ; shift ;;
+        --default-dir) DEFAULTDIR="$1" ; shift ;;
         --logs) LOGS_DIR="$1" ; shift ;;
         --spool) SPOOL_DIR="$1" ; shift ;;
         *) ARGS="$ARGS $PARAM" ;;
@@ -57,8 +58,11 @@
 if [ "$ETC_DIR" = "" ] ; then
     ETC_DIR={{sysconfdir}}/ejabberd
 fi
+if [ "$DEFAULTDIR" = "" ] ; then
+    DEFAULTDIR={{sysconfdir}}/default
+fi
 if [ "$EJABBERDCTL_CONFIG_PATH" = "" ] ; then
-    EJABBERDCTL_CONFIG_PATH=$ETC_DIR/ejabberdctl.cfg
+    EJABBERDCTL_CONFIG_PATH=$DEFAULTDIR/ejabberd
 fi
 if [ -f "$EJABBERDCTL_CONFIG_PATH" ] ; then
     . "$EJABBERDCTL_CONFIG_PATH"
@@ -265,7 +269,8 @@
     echo "Optional parameters when starting an ejabberd node:"
     echo "  --config-dir dir   Config ejabberd:    $ETC_DIR"
     echo "  --config file      Config ejabberd:    $EJABBERD_CONFIG_PATH"
-    echo "  --ctl-config file  Config ejabberdctl: $EJABBERDCTL_CONFIG_PATH"
+    echo "  --default file     Config ejabberdctl: $EJABBERDCTL_CONFIG_PATH"
+    echo "  --default-dir      Config ejabberdctl: $DEFAULTDIR"
     echo "  --logs dir         Directory for logs: $LOGS_DIR"
     echo "  --spool dir        Database spool dir: $SPOOL_DIR"
     echo "  --node nodename    ejabberd node name: $ERLANG_NODE"
